Refactor DbResults, DagFactory, Postgres, etc.

kevinji opened the pull request:
Created a few interfaces, converted Postgres to use String columns instead of int column indices, and other simplifications. This is the refactoring portion split off from #73.

kevinji committed:
editorconfig: Use 4-space indent for .cql

kevinji committed:
DbClient: Convert to an interface

kevinji committed:
DbResults: Create for Cassandra/Postgres

kevinji committed:
Fix compiler errors (refactoring)

kevinji committed:
Simplify Dag and DagFactory methods

maasg commented on src/main/java/edu/berkeley/ground/dao/models/cassandra/CassandraStructureVersionFactory.java:
```diff
@@ -115,25 +115,25 @@ public StructureVersion retrieveFromDatabase(long id) throws GroundException {
 
     if (attributesSet.isEmpty()) {
-      throw new GroundException("No StructureVersion attributes found for id " + id + ".");
+      throw new GroundException("No attributes found for StructureVersion with id " + id + ".");
```
maybe use `GroundVersionNotFoundException` instead?

maasg reviewed (commented):
+1 for increasing the abstraction and removing the specific "<backend>clients" from the calling classes. 
What would you think of making `DBClient` an `Iterable` ?  That would make simple `for-expressions` possible, instead of `do...while` and `while...` loops.
Like here: https://github.com/ground-context/ground/pull/75/files#diff-bf7526f106f74a35c573404ebb058024L135

kevinji committed:
StructureVersionFactory: Throw GroundVersionNotFoundException

kevinji committed:
Convert DbResults to an Iterable

kevinji committed:
Fix column order and bug in VersionSuccessor

Apparently the PostgreSQL API only works if you specify the columns
in the same order as the table that stores them.

kevinji committed:
Fix mocks, properly return null for DbResults's one()

kevinji committed:
cql/sql: Consistently capitalize commands

kevinji committed:
db: Add comments and license

kevinji committed:
Neo4jVersionHistoryDagFactory: Fix introduced bug

maasg commented on app/dao/models/cassandra/CassandraGraphVersionFactory.java:
```diff
@@ -132,16 +133,15 @@ public GraphVersion retrieveFromDatabase(long id) throws GroundException {
-        edgeVersionIds.add(edgeSet.getLong("edge_version_id"));
-      } while (edgeSet.next());
+    for (DbRow edgeRow : edgeSet) {
+      edgeVersionIds.add(edgeRow.getLong("edge_version_id"));
```
:+1:

vsreekanti commented on app/db/PostgresResults.java:
```diff
@@ -15,136 +15,194 @@
+      // Move cursor to row 1.
+      isEmpty = !this.resultSet.next();
     } catch (SQLException e) {
-      throw new GroundDbException(e);
```
Why wouldn't we want to throw an exception here?

kevinji replied:
I wasn't sure if constructors should throw an exception; I'll see if I can do that instead.

vsreekanti reviewed (approved):
This is awesome, @mc10 -- thanks!

kevinji committed:
PostgresResults: Have constructor throw exception

kevinji committed:
DaoTest: Slightly simplify
